easycord/_context_base.py

```
@@ -221,3 +221,26 @@ def get_member(self, user_id: int) -> discord.Member | None:
+        return await self.guild.fetch_member(user_id)
+
+    @property
+    def bot_permissions(self) -> discord.Permissions:
+        """The bot's own permissions in the current channel.
+
+        Returns ``channel.permissions_for(guild.me)`` — useful for checking
+        whether the bot can attach files, embed links, manage messages, etc.
+        Raises ``RuntimeError`` outside a guild.
+        """
+        if self.guild is None:
+            raise RuntimeError("bot_permissions requires a guild context")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)